pkg/input/validation.go

```diff
@@ -1,67 +1,64 @@
 package input
 
 import (
-	"regexp"
 	"strconv"
 	"strings"
 
+	valid "github.com/asaskevich/govalidator"
 	"github.com/pkg/errors"
 )
 
 type validatorIface interface {
 	validate(string) error
 }
 
-type nonEmptyValidator struct {
-	field *string
-}
+type nonEmptyValidator struct{}
 
 func (v nonEmptyValidator) validate(input string) error {
 	input = strings.TrimSpace(input)
 	if len(input) <= 0 {
 		return errors.New("empty command")
 	}
-	*v.field = input
+
 	return nil
 }
 
 type regexValidator struct {
 	pattern string
-	field   *string
 }
 
 func (v regexValidator) validate(input string) error {
 	input = strings.TrimSpace(input)
 	if len(input) <= 0 {
 		return errors.New("empty input")
 	}
-	pattern := regexp.MustCompile(v.pattern)
-	if !pattern.MatchString(input) {
-		return errors.New("invalid value")
+
+	if !valid.Matches(input, v.pattern) {
+		return errors.New("input does not matches pattern")
 	}
-	*v.field = input
+
 	return nil
 }
 
 type numericRangeValidator struct {
-	min   int
-	max   int
-	field *int
+	min int
+	max int
 }
 
 func (v numericRangeValidator) validate(input string) error {
 	input = strings.TrimSpace(input)
 	if len(input) <= 0 {
 		return errors.New("empty input")
 	}
-	port, err := strconv.Atoi(input)
+
+	num, err := strconv.Atoi(input)
 	if err != nil {
-		return errors.New("invalid port")
+		return errors.New("cannot convert input to a number")
```

Comment:
don't but negative/failure prefix into the error message - just the operation you want to perform

Reply:
I think `covert input to a number` is better here